com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/metadata/processor/EnvMtaMetadataParser.java

```diff
@@ -0,0 +1,134 @@
+package com.sap.cloud.lm.sl.cf.core.cf.metadata.processor;
+
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.sap.cloud.lm.sl.cf.core.Constants;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.ImmutableMtaMetadata;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadata;
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaResource;
+import com.sap.cloud.lm.sl.cf.core.model.ImmutableDeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.ImmutableDeployedMtaResource;
+import com.sap.cloud.lm.sl.common.ParsingException;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+import com.sap.cloud.lm.sl.mta.model.Version;
+
+@Component
+public class EnvMtaMetadataParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EnvMtaMetadataParser.class);
+    private static final List<String> ENV_METADATA_FIELDS = Arrays.asList(Constants.ENV_MTA_METADATA, Constants.ENV_MTA_MODULE_METADATA,
+                                                                         Constants.ENV_MTA_SERVICES,
+                                                                         Constants.ENV_MTA_MODULE_PUBLIC_PROVIDED_DEPENDENCIES);
+
+    public boolean hasMtaMetadata(CloudApplication application) {
+        return !Collections.disjoint(application.getEnv()
```

Comment:
This is a bit confusing. The check is if the keys of the env _don't have_ common elements with the metadata fields and then negates it. That's a bit roundabout.
I would prefer the other way around.
Change it to `app.getEnv().keySet().stream().anyMatch(ENV_METADATA_FIELDS::contains)`
And if you want that slight bit of performance, change the type of the env_metadata_fields to a HashSet

Reply:
Agreed. I also do not like the complexity of disjoint.